codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -160,6 +160,23 @@ private static boolean validateHeadersSentState(Http2Stream stream, Http2Headers
         return isInformational;
     }
 
+    /**
+     * According to RFC 9113, "Pseudo-header fields defined for requests MUST NOT appear in responses;
+     * pseudo-header fields defined for responses MUST NOT appear in requests."
+     * @return {@code true} if validation completed successfully, {@code false} if not
+     */
+    private boolean validatePseudoHeaders(Http2Headers headers) {
+        boolean isResponseHeaders = connection.isServer();
+        for (CharSequence name : headers.names()) {
+            if (Http2Headers.PseudoHeaderName.isPseudoHeader(name) &&
```

Comment:
this is super costy: try first to check if the prefix is a pseudoheader one

Reply:
The isPseudoHeader call eventually leads down to getPseudoHeaderName which has a check for the [prefix](https://github.com/netty/netty/blob/b8cddab13de5a83a1be48a94e5095388aa424007/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Headers.java#L131) so it should be okay.